app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -273,21 +275,56 @@ class ExplorationActivityPresenter @Inject constructor(
     explorationDataController.stopPlayingExploration(isCompletion).toLiveData()
       .observe(
         activity,
-        Observer<AsyncResult<Any?>> {
+        {
           when (it) {
             is AsyncResult.Pending -> oppiaLogger.d("ExplorationActivity", "Stopping exploration")
             is AsyncResult.Failure ->
               oppiaLogger.e("ExplorationActivity", "Failed to stop exploration", it.error)
             is AsyncResult.Success -> {
               oppiaLogger.d("ExplorationActivity", "Successfully stopped exploration")
-              backPressActivitySelector()
-              (activity as ExplorationActivity).finish()
+              if (isCompletion) {
+                shouldShowSurveyDialog(profileId, topicId)
```

Comment:
Perhaps instead: maybeShowSurveyDialog? The current name reads as a function that will return whether to show it as opposed to actually showing it.